eli5/keras.py

```diff
@@ -0,0 +1,293 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import numpy as np
+import keras
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Layer, Lambda
+from keras.preprocessing.image import load_img, img_to_array, array_to_img
+
+from eli5.base import Explanation
+from eli5.explain import explain_prediction
+
+
+DESCRIPTION_KERAS = """Grad-CAM visualization for image input; output is images"""
+
+# note that keras.models.Sequential subclasses keras.models.Model, so we can just register Model
+@explain_prediction.register(Model)
+def explain_prediction_keras(estimator, doc,
+                             target_names=None, # TODO: implement this
```

Comment:
Sorry if I'm repeating a question you already answered - how do you plan to implement this, where would these target names show up?

Reply:
I want to print, log, or add to the Explanation's `description` the predicted class that Grad-CAM focuses on.

Instead of printing the index of the class, I would print the actual label of the class. For example, for ImageNet with 1000 classes the user could pass a dictionary like [this](https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a) that we could then use to get the class label.

I think this is similar to what is done in this [tutorial](https://eli5.readthedocs.io/en/latest/tutorials/sklearn-text.html#baseline-model), where the feature names and labels are shown with this code:
```
eli5.show_weights(clf, vec=vec, top=10,
                  target_names=twenty_test.target_names)
```

Reply:
Thanks @teabolt , this makes sense 👍